cpp/src/packed/splitter/size_based_splitter.cpp

```diff
@@ -27,65 +27,60 @@ SizeBasedSplitter::SizeBasedSplitter(size_t max_group_size) : max_group_size_(ma
 
 void SizeBasedSplitter::Init() {}
 
-Result<std::vector<ColumnGroup>> SizeBasedSplitter::SplitRecordBatches(
+std::vector<ColumnGroup> SizeBasedSplitter::SplitRecordBatches(
     const std::vector<std::shared_ptr<arrow::RecordBatch>>& batches) {
-  auto schema = batches[0]->schema();
-
-  ASSIGN_OR_RETURN_ARROW_NOT_OK(auto merged_table, arrow::Table::FromRecordBatches(schema, batches));
-
-  std::vector<std::shared_ptr<Array>> arrays;
-  for (const auto& column : merged_table->columns()) {
-    // Concatenate all chunks of the current column
-    ASSIGN_OR_RETURN_ARROW_NOT_OK(auto concatenated_array,
-                                  arrow::Concatenate(column->chunks(), arrow::default_memory_pool()));
-    arrays.push_back(concatenated_array);
-  }
-  std::shared_ptr<RecordBatch> batch =
-      RecordBatch::Make(merged_table->schema(), merged_table->num_rows(), std::move(arrays));
-  LOG_STORAGE_INFO_ << "split record batch: " << merged_table->num_rows();
-  return Split(batch);
-}
-
-std::vector<ColumnGroup> SizeBasedSplitter::Split(const std::shared_ptr<arrow::RecordBatch>& record) {
-  if (!record) {
-    throw std::invalid_argument("RecordBatch is null");
+  if (batches.empty()) {
+    return {};
   }
-  std::vector<ColumnGroup> column_groups;
-  std::vector<int> small_group_indices;
-  GroupId group_id = 0;
-  for (int i = 0; i < record->num_columns(); ++i) {
-    std::shared_ptr<arrow::Array> column = record->column(i);
-    if (!column) {
-      throw std::runtime_error("Column is null");
+  // calculate column sizes and rows
+  std::vector<size_t> column_sizes(batches[0]->num_columns(), 0);
+  std::vector<int64_t> column_rows(batches[0]->num_columns(), 0);
+  for (const auto& record : batches) {
+    for (int i = 0; i < record->num_columns(); ++i) {
+      std::shared_ptr<arrow::Array> column = record->column(i);
+      if (!column) {
+        throw std::runtime_error("Column is null");
```

Comment:
The error handling is supposed to be either exception or return value of type `Result`, not both. BTW, I don't think it's necessary to check null here.